src/lib.rs

```diff
@@ -2,15 +2,24 @@
 #![allow(dead_code)]
 #![cfg_attr(docsrs, feature(doc_cfg))]
 
-#[cfg(not(any(feature = "postgres", feature = "mysql", feature = "sqlite",)))]
-compile_error!("one of the features ['postgres', 'mysql', 'sqlite'] must be enabled");
+#[cfg(not(any(
+    feature = "postgres",
+    feature = "mysql",
+    feature = "sqlite",
+    features = "DatabaseTrait"
+)))]
+compile_error!(
+    "one of the features ['postgres', 'mysql', 'sqlite','DatabaseTrait'] must be enabled"
+);
 
 #[cfg(any(
     all(feature = "postgres", feature = "mysql"),
     all(feature = "postgres", feature = "sqlite"),
+    all(feature = "postgres", feature = "DatabaseTrait"),
+    all(feature = "mysql", feature = "DatabaseTrait"),
```

Comment:
need to add
all(feature = "sqlite", feature = "DatabaseTrait"),

Though we should review this better as we might be able to remove this Completely since we are enforcing a trait now.